src/pfe/portal/modules/User.js

```diff
@@ -639,54 +639,51 @@ module.exports = class User {
   /**
    * Function to write workspace settings
    */
-  async writeWorkspaceSettings(workspaceSettings) {
+  async writeWorkspaceSettings(address, namespace) {
     let retval;
     try {
       log.info(`Writing workspace settings file.`);
-      retval = await this.fw.writeWorkspaceSettings(workspaceSettings);
+      retval = await this.fw.writeWorkspaceSettings(address, namespace);
     } catch (err) {
       log.error(`Error in writeWorkspaceSettings`);
-      log.error(err);
+      throw err;
     }
     log.debug(`writeWorkspaceSettings return value: ` + JSON.stringify(retval));
     return retval;
   }
 
   /**
-   * Function to get test deployment registry
+   * Function to get test image push registry
    */
-  async testDeploymentRegistry(deploymentRegistry) {
+  async testImagePushRegistry(address, namespace) {
     let retval;
     try {
-      retval = await this.fw.testDeploymentRegistry(deploymentRegistry);
+      retval = await this.fw.testImagePushRegistry(address, namespace);
     } catch (err) {
-      log.error(`Error in testDeploymentRegistry`);
-      log.error(err);
+      log.error(`Error in testImagePushRegistry`);
+      throw err;
     }
     return retval;
   }
 
   /**
    * Function to setup the Docker config file and Kube secret
    */
-  async setupRegistrySecret(credentials, url) {
+  async setupRegistrySecret(credentials, address) {
     const credentialsJSON = JSON.parse(Buffer.from(credentials, "base64").toString());
     if (credentialsJSON.username == undefined || credentialsJSON.password == undefined) {
-      throw new RegistrySecretsError("INVALID_ENCODED_CREDENTIALS", "for url " + url);
+      throw new RegistrySecretsError("INVALID_ENCODED_CREDENTIALS", "for address " + address);
     }
     const username = credentialsJSON.username;
     const password = credentialsJSON.password;
-    log.info("Setting up the Docker Registry secret for url: " + url + " and username: " + username);
+    log.info("Setting up the Docker Registry secret for address: " + address + " and username: " + username);
     const registrySecretList = [];
     
     try {
       // Handle dockerhub specifically on local since docker.io does not work
-      if (url === "docker.io" && !global.codewind.RUNNING_IN_K8S) {
+      if (address === "docker.io" && !global.codewind.RUNNING_IN_K8S) {
         // eslint-disable-next-line no-param-reassign
-        url = "https://index.docker.io/v1/";
-      } else if (!url.startsWith("https://") && !url.startsWith("http://")) {
-        // eslint-disable-next-line no-param-reassign
-        url = "https://" + url ;
+        address = "https://index.docker.io/v1/";
```

Comment:
Should this be ended with / ? I think in one of the places we are doing address + '/' + namespace.

Reply:
yes we're doing that when we read in address `imagePushRegistry = workspaceSettingsInfo.registryAddress.replace(/\/\s*$/, "") + "/" + workspaceSettingsInfo.registryNamespace.trim();`